integration-tests/opentelemetry/src/test/java/io/quarkus/it/opentelemetry/MetricsTest.java

```diff
@@ -54,7 +62,11 @@ public void directCounterTest() {
                 .then()
                 .statusCode(200);
 
-        await().atMost(5, SECONDS).until(() -> getSpans().size() == 2);
+        await().atMost(10, SECONDS).until(() -> {
+            List<Map<String, Object>> spans = getSpans();
+            System.out.println("spans size " + spans.size());
```

Comment:
Maybe you should display the content of `spans` instead? That might be more helpful if we have to debug it.

Reply:
Kind of... Most likely it was failing because it was also reading a span from a previous test, throwing the count to 3.
I just want make sure, if it happens again.